Assets/HoloToolkit/UX/Scripts/Lines/LineUnity.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using HoloToolkit.Unity;
+using System.Collections;
+using UnityEngine;
+
+namespace Holotoolkit.Unity.UX
+{
+    public class LineUnity : LineRendererBase
+    {
+        const string DefaultLineShader = "Particles/Alpha Blended";
+        const string DefaultLineShaderColor = "_TintColor";
+
+        [Header("LineUnity Settings")]
+        [Tooltip("The material to use for the Unity LineRenderer (will be auto-generated if null)")]
+        public Material LineMaterial;
+
+        public bool RoundedEdges = true;
+        public bool RoundedCaps = true;
+
+        protected override void OnEnable()
+        {
+            base.OnEnable();
+
+            if (lineRenderer == null)
+                lineRenderer = gameObject.GetComponent<UnityEngine.LineRenderer>();
+            if (lineRenderer == null)
+                lineRenderer = gameObject.AddComponent<UnityEngine.LineRenderer>();
+
+            if (LineMaterial == null) {
+                LineMaterial = new Material(Shader.Find(DefaultLineShader));
+                LineMaterial.SetColor(DefaultLineShaderColor, Color.white);
+            }
+
+            StartCoroutine(UpdateLineUnity());
+        }
+
+        private IEnumerator UpdateLineUnity()
+        {
+            while (isActiveAndEnabled)
+            {
+                if (!source.enabled)
+                {
+                    lineRenderer.enabled = false;
+                }
+                else
+                {
+                    lineRenderer.enabled = true;
+
+                    switch (this.StepMode)
```

Comment:
why call `this.StepMode` are we hiding an inherited property?

Reply:
Legacy cruft. Removed.